[[Documentation]] keyword/long - detailed|long

seaniepie opened the pull request:
- camelCase from defaultfolder to defaultFolder
- Describe relationship between 'long' and 'detailed' with script formatting.

seaniepie committed:
[[Documentation]] keyword/long - detailed|long

- camelCase from defaultfolder to defaultFolder
- Describe relationship between 'long' and 'detailed' with script formatting.

seaniepie commented:
There seems to be a mismatch in references to the 'long' keyword in the files and folders functions documentation. There is also no link from them to 'file(keyword)' and 'folder(keyword)' and vice versa. Does this need to be addressed?

livecodeali commented:
@seaniepie If by a mismatch you mean that the `long` keyword should be referenced in the `files` and `folders` function entries, then yes I agree.

The `file` and `folder` keywords probably shouldn't have links to the `long` keyword, as they are referring to file and folder in the sense of `there is a file | folder` which doesn't relate to the `long` keyword.

seaniepie commented:
1. There is not a 'Folder' keyword dictionary entry. Does this mean that the word folder is not a keyword even when used in the `there is a folder` form.
2. 'File' keyword is only referred to in the lcdoc for it in reference to `URL "file:*.*"`, again not for `there is a file`.
3. What I meant, to clarify, is that in both the 'function/files' and 'function/folders' docs there is no reference 'linking' to this 'keyword/long' doc or 'detailed'. You've answered that tho so I'll fix both of those too.

livecodeali commented:
@seaniepie Ah yes, right you are!

seaniepie commented:
@seaniepie Link files and folders to this and vice versa

seaniepie committed:
[Docs] files - Add 'long' keyword references

seaniepie committed:
[Docs] folders - Add 'long' keyword references

seaniepie committed:
[Docs] long - Add links to Date, Time, Name, ID & useSystemDate

seaniepie committed:
[Docs] ID - Add references to 'long' keyword

seaniepie committed:
[Docs] Name - Added links to the 'Long' keyword

seaniepie committed:
[Docs] Date - Add links to the 'long' keyword

seaniepie committed:
[Docs] Time - Add links to 'Long' keyword

seaniepie commented:
@livecodeali , the time doc has references to `<format|formatted>` but it has links to both the function and glossary in the references section. Does the tag need defining or does it go to the first link by default?

livecodeali commented:
@seaniepie The `format` function is the one that creates strings with substituted parameters and so on, which I think is completely irrelevant to the `time` function, so that related item should be removed entirely. All uses of the word `format` are just to do with the concept of format rather than the function as far as I can tell!

seaniepie committed:
Update - Remove reference to the 'format' func

The term 'format' here should only refer to the glossary as its use only relates to the concept of a Format rather than the LiveCode function.

seaniepie commented:
👍

livecodeali commented on docs/dictionary/keyword/long.lcdoc:
```diff
@@ -26,36 +26,39 @@ Example:
 looks like this: card "My Card" of stack "/Drive/Folder/Stack.rev"
 
-A long object ID looks like this:
+A long object <ID> looks like this:
```
These are case sensitive IIRC.

livecodeali replied:
This should be directed at the instance in the references list, sorry 😄

livecodeali commented on docs/dictionary/keyword/long.lcdoc:
```diff
@@ -69,7 +72,8 @@ propertyNames (function), time (function), folders (function),
 keyword (glossary), property (glossary), command (glossary),
 object type (glossary), expression (glossary), object (glossary),
-card (keyword), short (keyword), abbreviated (keyword), owner (property)
+card (keyword), short (keyword), abbreviated (keyword), id (property)
```
This needs to be `ID` rather than `id`.

seaniepie replied:
Got it.

seaniepie committed:
update - Reference for ID was in lower case. Now upper.

livecodeali commented:
@livecode-vulcan review ok 897cec7

livecodeali commented:
Thanks @seaniepie!

livecode-vulcan commented:
💙 review by @livecodeali ok 897cec717bb862af3f1eb2a0b4f6fb971be5d980

livecode-vulcan commented:
😞 test failure 897cec717bb862af3f1eb2a0b4f6fb971be5d980
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/503)
- try-community-mac: [pending](https://vulcan.livecode.com/builders/try-community-mac/builds/508)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/503)
- try-community-emscripten: [failure](https://vulcan.livecode.com/builders/try-community-emscripten/builds/503)
- try-community-ios: [pending](https://vulcan.livecode.com/builders/try-community-ios/builds/509)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/503)
- try-community-android-armv6: [failure](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/503)

seaniepie commented:
Has this emscripten/android fail been fixed now. I saw @montegoulding was working on it today.

montegoulding commented:
I think @peter-b is working on it and it's still a thing

seaniepie commented:
@montegoulding Travis should have picked up on something in the 'files' doc. In the syntax we have a parameter, `<targetFolder>`, which is not in a parameters section. I'm adding a param section to it now but it's interesting that CI wasn't able to pick it up.

seaniepie committed:
update - Example array and missing param

- In the last Example the array name did not follow convention. (Personally I prefer to also add a capital A to the end of the name so I know it is used as an Array)
- Also added the missing parameter, targetFolder

seaniepie committed:
update folders - Array name change and add parameter

- Array name in last example was not following naming convention
- Parameter `targetFolder` added.

seaniepie commented:
Folders doc had the same issue.

montegoulding commented:
@seaniepie the develop branch doesn't have the full range of tests that my validatedocs branch does. There are errors about this on the validatedocs branch I believe.

montegoulding commented:
@livecode-vulcan review ok 24aa3d8

livecode-vulcan commented:
💙 review by @montegoulding ok 24aa3d84fa3e54a2b8898240a7b187b8e3a2ee15

livecode-vulcan commented:
😞 test failure 24aa3d84fa3e54a2b8898240a7b187b8e3a2ee15
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/522)
- try-community-mac: [pending](https://vulcan.livecode.com/builders/try-community-mac/builds/527)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/522)
- try-community-emscripten: [failure](https://vulcan.livecode.com/builders/try-community-emscripten/builds/526)
- try-community-ios: [pending](https://vulcan.livecode.com/builders/try-community-ios/builds/531)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/523)
- try-community-android-armv6: [failure](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/523)

montegoulding commented:
@livecode-vulcan try 24aa3d8

livecode-vulcan commented:
😎 test success 24aa3d84fa3e54a2b8898240a7b187b8e3a2ee15
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/548)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/553)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/548)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/552)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/557)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/549)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/549)
